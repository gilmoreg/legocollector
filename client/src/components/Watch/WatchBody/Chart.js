import React from 'react';
import PropTypes from 'prop-types';
// https://github.com/jerairrest/react-chartjs-2/blob/master/example/src/components/line.js
import { Line } from 'react-chartjs-2';
import moment from 'moment';
import './Chart.css';

const Chart = (props) => {
  const data = {
    labels: props.stock_levels.map(level =>
      moment(new Date(level.datetime)).format('MM-DD')),
    datasets: [
      {
        label: false,
        fill: false,
        lineTension: 0.1,
        backgroundColor: 'rgba(75,192,192,0.4)',
        borderColor: 'rgba(75,192,192,1)',
        borderCapStyle: 'butt',
        borderDash: [],
        borderDashOffset: 0.0,
        borderJoinStyle: 'miter',
        pointBorderColor: 'rgba(75,192,192,1)',
        pointBackgroundColor: '#fff',
        pointBorderWidth: 1,
        pointHoverRadius: 5,
        pointHoverBackgroundColor: 'rgba(75,192,192,1)',
        pointHoverBorderColor: 'rgba(220,220,220,1)',
        pointHoverBorderWidth: 2,
        pointRadius: 1,
        pointHitRadius: 10,
        data: props.stock_levels.map(level => level.stock_level),
      },
    ],
  };
  const options = {
    legend: {
      display: false,
    },
  };

  return (
    <div className="Chart">
      <Line data={data} options={options} />
    </div>
  );
};

Chart.propTypes = {
  stock_levels: PropTypes.arrayOf(PropTypes.shape({
    datetime: PropTypes.string,
    id: PropTypes.number,
    stock_level: PropTypes.number,
  })).isRequired,
};

export default Chart;
