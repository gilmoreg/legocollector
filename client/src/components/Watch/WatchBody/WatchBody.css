.WatchBody {
    border-left: 1px solid black;
    border-right: 1px solid black;
    max-width: 760px;
}

.WatchBody .graph {
    width: 100%;
    height: 150px;
    margin: 0 auto;
    padding: 25px;
}

.WatchBody .notifications {
    display: flex;
    justify-content: center;
    align-content: space-around;
    padding-bottom: 50px;
}

.WatchBody .label {
    flex: 0 0 20%;
    align-self: center;
}


.WatchBody .slider {
    flex: 0 0 30%;
    -webkit-appearance: none;  /* Override default CSS styles */
    appearance: none;
    width: 50%; /* Full-width */
    height: 25px; /* Specified height */
    background: #d3d3d3; /* Grey background */
    outline: none; /* Remove outline */
    opacity: 0.7; /* Set transparency (for mouse-over effects on hover) */
    -webkit-transition: .2s; /* 0.2 seconds transition on hover */
    transition: opacity .2s;
}

/* Mouse-over effects */
.WatchBody .slider:hover {
    opacity: 1; /* Fully shown on mouse-over */
}

/* The slider handle (use webkit (Chrome, Opera, Safari, Edge) and moz (Firefox) to override default look) */ 
.WatchBody .slider::-webkit-slider-thumb {
    -webkit-appearance: none; /* Override default look */
    appearance: none;
    width: 25px; /* Set a specific slider handle width */
    height: 25px; /* Slider handle height */
    background: #4CAF50; /* Green background */
    cursor: pointer; /* Cursor on hover */
}

.WatchBody .slider::-moz-range-thumb {
    width: 25px; /* Set a specific slider handle width */
    height: 25px; /* Slider handle height */
    background: #4CAF50; /* Green background */
    cursor: pointer; /* Cursor on hover */
}

.WatchBody .value {
    flex: 0 0 5%;
    align-self: center;
    padding: 0 25px 0 25px;
}