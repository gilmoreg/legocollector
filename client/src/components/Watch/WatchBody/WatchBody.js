import React from 'react';
import PropTypes from 'prop-types';
import Chart from './Chart';

import './WatchBody.css';

const WatchBody = props => (
  <div className="WatchBody">
    <Chart stock_levels={props.stock_levels} />
    <button className="delete" onClick={props.deleteWatchClick}>
      <i className="fa fa-times" aria-hidden="true" />
    </button>
  </div>
);

WatchBody.propTypes = {
  stock_levels: PropTypes.arrayOf(PropTypes.shape({
    datetime: PropTypes.string,
    id: PropTypes.number,
    stock_level: PropTypes.number,
  })).isRequired,
  deleteWatchClick: PropTypes.func.isRequired,
};

export default WatchBody;
