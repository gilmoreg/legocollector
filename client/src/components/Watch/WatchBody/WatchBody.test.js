import React from 'react';
import { shallow } from 'enzyme';
import WatchBody from './WatchBody';

it('renders without crashing', () => {
  const stockLevels = [
    { datetime: 'test', stock_level: 1 },
    { datetime: 'test', stock_level: 1 },
  ];
  const wrapper = shallow(
    <WatchBody
      stock_levels={stockLevels}
      deleteWatchClick={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
