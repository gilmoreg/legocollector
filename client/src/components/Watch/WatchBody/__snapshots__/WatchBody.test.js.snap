// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchBody
    deleteWatchClick={[Function]}
    stock_levels={
      Array [
        Object {
          "datetime": "test",
          "stock_level": 1,
        },
        Object {
          "datetime": "test",
          "stock_level": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Chart
          stock_levels={
            Array [
              Object {
                "datetime": "test",
                "stock_level": 1,
              },
              Object {
                "datetime": "test",
                "stock_level": 1,
              },
            ]
          }
        />,
        <button
          className="delete"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>,
      ],
      "className": "WatchBody",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "stock_levels": Array [
            Object {
              "datetime": "test",
              "stock_level": 1,
            },
            Object {
              "datetime": "test",
              "stock_level": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <i
            aria-hidden="true"
            className="fa fa-times"
          />,
          "className": "delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-times",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Chart
            stock_levels={
              Array [
                Object {
                  "datetime": "test",
                  "stock_level": 1,
                },
                Object {
                  "datetime": "test",
                  "stock_level": 1,
                },
              ]
            }
          />,
          <button
            className="delete"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-times"
            />
          </button>,
        ],
        "className": "WatchBody",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "stock_levels": Array [
              Object {
                "datetime": "test",
                "stock_level": 1,
              },
              Object {
                "datetime": "test",
                "stock_level": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <i
              aria-hidden="true"
              className="fa fa-times"
            />,
            "className": "delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-times",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
